specification/ai/Azure.AI.Projects/agents/runs/models.tsp

```diff
@@ -143,6 +143,7 @@ model ThreadRun {
   toolResources?: UpdateToolResourcesOptions | null;
 
   @doc("Determines if tools can be executed in parallel within the run.")
+  @encodedName("application/json", "parallel_tool_calls")
```

Comment:
Is @encodedName really needed here? Isn't the default already good? Meaning TypeSpec variable named parallelToolCalls gets converted to JSON element name parallel_tool_calls. You can check that by removing the @encodedName, compiling the TypeSpec and looking at the resulting JSON swagger

Reply:
Yes, it looks like code works without this line; removed.

Reply:
You removed one instance of `@encodedName("application/json", "parallel_tool_calls")` but I see two more. Can't they also be removed?

Reply:
When I remove @encodedName in option class it does not work. Here are my findings:

 Generated code does not convert camelCase variables to snake_case and that is why request with "parallelToolCalls" will fail - it is not present in spec and server will return 400.
However, if the ThreadRun's parallelToolCalls is missing encodedName , everything work. That happens, because in python the absence of annotation will create code "parallel_tool_calls: bool = rest_field(name="parallelToolCalls")". When the JSON will be deserialized the _RestField class will be used to apply appropriate formatting to rest field. First we ⁠[search field](https://github.com/Azure/azure-sdk-for-python/blob/347b6cbc1fda32568357d3ebf6f9ddc3f881e034/sdk/ai/azure-ai-projects/azure/ai/projects/_model_base.py#L476) by name and, when it will fail, we just give to the constructor whatever ⁠[we have got from server](https://github.com/Azure/azure-sdk-for-python/blob/347b6cbc1fda32568357d3ebf6f9ddc3f881e034/sdk/ai/azure-ai-projects/azure/ai/projects/_model_base.py#L483). So this code is, essentially silently failing without side effects. 

So we always need to mark the fields unless the field name is the same as its name in the JSON.

Reply:
Understood! thank you for checking this out.